### PR TITLE
Get messages referenced in a thread

### PR DESCRIPTION
**Pull request by dominictarr**
The main effect of this is to pull in old threads that where commented on recently.
sometimes this will retrieve messages twice, because it requests a message, then then it comes in the stream. to fix this, I think just slow down the get, for say half a second.

### CODE REVIEW

File: state/obs.js [hunk 1]
```diff
@@ -52,6 +53,23 @@ exports.create = function (api) {
...
 
+    var getting = {}, g= 0, t = 0
+    var maybe = {}
+    obs(function (state) {
+      if(state.effect) {
+        var effect = state.effect
+        state.effect = null
+        if(!getting[effect.key]) {
+          getting[effect.key] = true
+          api.sbot.async.get(effect.key, function (err, msg) {-            console.log('g', g, ++t)
```
Comment (mixmix) [inline state/obs.js]: I've removed this console.log and that rogue `-`
I've changed this from a nested if to an early-return pattern to try and understand the logic.

Only changes logic from 
```js
if (A) {
  if (B) {
   // do things
  }
}
```
to :
```js
if (!A) return 
if (!B) return
// do things
```

**Comment by mixmix** [comment]
Sweet, sucessfully loads private threads in home view now.

Still seeing double `#cjdns` alas

**Comment by dominictarr** [comment]
maybe that is a one is a forked thread or something?

**Comment by dominictarr** [comment]
I'm definitely still seeing some threads as incomplete, I'm guessing caused by forking

**Pull request merged**